packages/flame/lib/src/utils/recycled_queue.dart

```diff
@@ -0,0 +1,120 @@
+import 'dart:math';
+
+/// [RecycledQueue] is a simple FIFO queue where the elements are recycled.
+///
+/// That is, the elements [T] in this queue are created and owned by the queue
+/// and will not get destroyed when removed from the queue. Instead, these
+/// objects will be `dispose()`-d and then released into the pool of unused
+/// elements. When new objects are added to the queue, they will be retrieved
+/// from the pool if available.
+///
+/// The API of this class is slightly different from the traditional Queue:
+/// - [addLast] appends a new element to the end of the queue and then returns
+///   it to the user to fill.
+/// - [removeFirst] deletes and disposes of the first element without returning
+///   it (use [first] to retrieve the first element beforehand).
+///
+/// Internally, the queue is backed by a circular list.
+class RecycledQueue<T extends Disposable> {
+  RecycledQueue(this.factory, {int initialCapacity = 8})
+      : assert(initialCapacity > 0),
+        _elements = List.generate(initialCapacity, (i) => factory()),
+        _startIndex = -1,
+        _endIndex = -1;
+
+  /// Function for creating new elements in the queue.
+  final T Function() factory;
+
+  /// Index of the first element in the queue, or -1 if the queue is empty.
+  int _startIndex;
+
+  /// Index of the last element in the queue, or -1 if the queue is empty. Note
+  /// that this index is inclusive.
+  int _endIndex;
+
+  /// The backing container.
+  ///
+  /// Some of the items in this list are considered "active" while others are
+  /// "disposed". The [_startIndex] and [_endIndex] describe the range of items
+  /// in the list which are "active".
+  ///
+  /// Two data layouts are possible: the normal one, when [_endIndex] is larger
+  /// than or equal to the [_startIndex]:
+  ///   [----##############--]
```

Comment:
Do these not have to be in back ticks for the doc parser not to regard them as references to fields?

Reply:
I've added the backticks.